library/cipher.c

```diff
@@ -1161,11 +1161,8 @@ int mbedtls_cipher_check_tag( mbedtls_cipher_context_t *ctx,
     }
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
-    /* Status to return on a non-authenticated algorithm. It would make sense
-     * to return MBEDTLS_ERR_CIPHER_INVALID_CONTEXT or perhaps
-     * MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA, but at the time I write this our
-     * unit tests assume 0. */
-    ret = 0;
+    /* Status to return on a non-authenticated algorithm. */
```

Comment:
While the removed text suggests using `MBEDTLS_ERR_CIPHER_INVALID_CONTEXT` or `MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA`, [the original PR](https://github.com/Mbed-TLS/mbedtls/pull/2164) uses `MBEDTLS_ERR_CIPHER_FEATURE_UNAVAILABLE` [as recommended by Gilles in a comment](https://github.com/Mbed-TLS/mbedtls/pull/2164#discussion_r231119704)